From 8a6812ae004a43aad29604e86d7c746009d9d2c9 Mon Sep 17 00:00:00 2001
From: Icenowy Zheng <uwu@icenowy.me>
Date: Thu, 4 Apr 2024 07:39:13 +0800
Subject: [PATCH] fix Linux installation creating broken
 superslicer-gcodeviewer symlink

Currently the symlink target is hardcoded as slic3r, which is not the
binary name of SuperSlicer.

Change it to use the variable for application command name instead.

Signed-off-by: Icenowy Zheng <uwu@icenowy.me>
---
 src/CMakeLists.txt | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index 90a891c70..76a0b495e 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -301,5 +301,5 @@ else ()
     install(TARGETS Slic3r RUNTIME DESTINATION "${CMAKE_INSTALL_BINDIR}")
 
     # Install the symlink for gcodeviewer
-    install(CODE "execute_process(COMMAND ${CMAKE_COMMAND} -E create_symlink slic3r ${GCODEVIEWER_APP_CMD} WORKING_DIRECTORY \$ENV{DESTDIR}/${CMAKE_INSTALL_PREFIX}/${CMAKE_INSTALL_BINDIR})")
+    install(CODE "execute_process(COMMAND ${CMAKE_COMMAND} -E create_symlink ${SLIC3R_APP_CMD} ${GCODEVIEWER_APP_CMD} WORKING_DIRECTORY \$ENV{DESTDIR}/${CMAKE_INSTALL_PREFIX}/${CMAKE_INSTALL_BINDIR})")
 endif ()
-- 
2.44.0

